core/src/Lender.sol

```diff
@@ -98,16 +106,16 @@ contract Lender is Ledger {
             if (allowed != type(uint256).max) allowance[owner][msg.sender] = allowed - shares;
         }
 
-        // Transfer tokens
-        cache.lastBalance -= amount;
-        asset().safeTransfer(recipient, amount);
-
-        // Burn shares (emits event that can be interpreted as a withdrawal)
-        _unsafeBurn(owner, shares);
+        // Burn shares and (if applicable) handle courier accounting
+        amount = _unsafeBurn(owner, shares, amount);
         unchecked {
             cache.totalSupply -= shares;
         }
 
+        // Transfer tokens
+        cache.lastBalance -= amount;
+        asset().safeTransfer(recipient, amount);
```

Comment:
Moved this logic down below the `_unsafeBurn` since `_unsafeBurn` now updates `amount`